File: Data Structures & Algorithms/combination-target-sum/submission-3.java
class Solution {

    private void rec(int i,int n,List<Integer> temp,List<List<Integer>> ans,int[] nums,int target)
    {
        if(target < 0)
            return;
        
        if(target == 0)
        {
            ans.add(new ArrayList<>(temp));
            return;
        }
        if(i == n)
        {
            return;
        }

        //Take and stay at same index
        temp.add(nums[i]);
        rec(i,n,temp,ans,nums,target - nums[i]);
        temp.remove(temp.size() - 1);
        rec(i+1,n,temp,ans,nums,target);
    }
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        rec(0,nums.length,temp,ans,nums,target);
        return ans;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        l,r = 0,n-1
        ans = []

        while l < r:
            total = nums[l] + nums[r]
            if total == target:
                ans.append(l+1)
                ans.append(r+1)
                return ans
            elif total > target:
                r -= 1
            else:
                l += 1